Lib/contextlib.py

```diff
@@ -365,49 +357,28 @@ def pop_all(self):
         self._exit_callbacks = deque()
         return new_stack
 
-    def _push_cm_exit(self, cm, cm_exit):
-        """Helper to correctly register callbacks to __exit__ methods"""
-        def _exit_wrapper(*exc_details):
-            return cm_exit(cm, *exc_details)
-        _exit_wrapper.__self__ = cm
-        self.push(_exit_wrapper)
-
-    def push(self, exit):
+    def push(self, exit_obj):
```

Comment:
Folks may be calling `stack.push(exit=obj)`, so we shouldn't gratuitously change parameter names.